all-sdk/external-client-configuration.md

```diff
@@ -0,0 +1,526 @@
+    temporal config set --profile dev --key tls.client_key_path --value path/to/my/dev-cert.key
+
+And prod:
+
+    temporal config set --profile prod --key address --value my-prod-ns.a1b2c.tmprl.cloud:7233
+    temporal config set --profile prod --key namespace --value my-prod-ns.a1b2c
+    temporal config set --profile prod --key tls.client_cert_path --value path/to/my/prod-cert.pem
+    temporal config set --profile prod --key tls.client_key_path --value path/to/my/prod-cert.key
+
+Now a simple setting of `TEMPORAL_PROFILE` environment to `dev` or `prod` will switch connectivity (or `--profile` on
+CLI or the string in the SDK if you don't want to use environment variable).
+
+#### Possible UI Creation Flow
+
+When a user creates an API key, the completion screen can provide a download of the config file that will have the
+address, namespace, and API key set. So it can say something like:
+
+```
+API key is created, make sure you copy it here because it is not stored:
+
+    <some long api key string>
+
+Alternatively, you can download it as a configuration file:
+
+    <some button or something that downloads a file name temporalio-client.toml>
+
+You can place this configuration file in the default place used by CLIs/SDKs:
+
+* macOS - `$HOME/Library/Application Support/temporalio/temporal-client.toml`
+* Windows - `%AppData%/temporalio/temporal-client.toml` (often `C:\Users\<myuser>\AppData\Roaming\temporalio\temporal-client.toml`)
+* Linux - `$HOME/.config/temporalio/temporal-client.toml`
+
+This will be automatically consumed by the SDKs/CLI. This file can also be put anywhere else and `TEMPORAL_CONFIG_FILE`
+environment variable can be set to point to it.
+```
+
+#### Possible CLI Creation Flow
+
+Assuming cloud support is present in `temporal` CLI, you can have the normal certificate generation steps, e.g.
+
+```
+temporal cloud gen ca ...
+temporal cloud gen leaf ...
+temporal cloud namespace add-accepted-client-ca ...
+```
+
+And that last step can by default set a `temporal-cloud` profile in the configuration with the address, namespace, and
+mTLS certificates. So now all you have to do to use cloud in CLI is either `--profile temporal-cloud` or set
+`TEMPORAL_PROFILE=temporal-cloud` environment variable. Granted you may want better certificate management than that and
+whether this is done by default can be discussed, but the idea is the same.
+
+#### Multi-Namespace Config Use
+
+Some users may want the same configuration for multiple namespaces (or multiple addresses or multiple API keys or
+whatever). Users can supply partial configuration. For instance, you can have this config:
+
+```toml
+[profile.temporal-cloud]
+address = "my-dev-ns.a1b2c.tmprl.cloud:7233"
+tls.client_cert_path = "path/to/my/dev-cert.pem"
+tls.client_key_path = "path/to/my/dev-cert.pem"
+```
+
+And then still use CLI like
+
+    temporal workflow list --namespace my-specific-namespace
```

Comment:
Uh? Why is this implicitly using the `temporal-cloud` profile?

Did you mean the profile above to be named `default` instead? Or add a `--profile temporal-cloud` argument to this command and the one below?

Reply:
Also, would this work on Cloud _today_? or only once we get global endpoints?

If not, I think we should use a non-cloud example, or clearly indicate that this is pending future Cloud changes.

Reply:
> Did you mean the profile above to be named default instead?

Yes, sorry, when translating to TOML in https://github.com/temporalio/proposals/pull/95/commits/b8cf8febcb7c68f4a54415e7b111428994cfc8eb I accidentally change the profile name, needs to be `default`, will fix

Reply:
> Also, would this work on Cloud today? or only once we get global endpoints?

Yes for data plane. Clients are just addresses and namespaces and auth. What shared endpoints will provide is a way for cloud ops API to use this as well.